homeassistant/components/govee_light_api/light.py

```diff
@@ -0,0 +1,213 @@
+"""Govee Light."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+import logging
+from typing import Any
+
+from govee_local_api import GoveeDevice
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP_KELVIN,
+    ATTR_RGB_COLOR,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SCAN_INTERVAL
+from .capability import (
+    GOVEE_COORDINATORS_MAPPER,
+    GOVEE_DEVICE_CAPABILITIES,
+    GoveeLightCapabilities,
+)
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import GoveeLocalApiCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Govee light setup."""
+
+    entry_id = config_entry.entry_id
+    coordinator: GoveeLocalApiCoordinator = hass.data.setdefault(DOMAIN, {}).get(
+        entry_id, None
+    )
```

Comment:
Can we move this to `__init__.py`?

Reply:
Do you mean to move the entire code of async_setup_entry in the one in the __init__.py?

Reply:
We want to create the coordinator in the init file, not in a platform file